web_app/calendar/app.py

```diff
@@ -1,13 +1,36 @@
 import copy
-from flask import Flask
+from flask import Flask, request, redirect
 
 from helpers import gen_hours, gen_weeks, date_template
 from html_template import *
 
-
 app = Flask(__name__)
 
 
+@app.route(task_preset_link)
+@app.route(task_preset_link + '/')
+def get_form():
+    args = dict(request.args)
+    form = task_preset_form.format(redirect=args.get('redirect')[0],
```

Comment:
Direct call for http://localhost:5000/task_preset/ leads to an error

Reply:
Additionally, when this link redirects to another page (like this - http://localhost:5000/task_preset?y=2018&m=5&d=15&h=9&redirect=http://localhost:5000/calendar/years/2018/months/5/days/15/hours/), status code is 301. I think 302 is more appropriate code.

Reply:
I can drop one of route but it'll be less cozy.

Reply:
"One of __the routes__", because you have more than one route and choose between them.